### PR TITLE
community: Add ITREX optimized Embeddings 

### PR DESCRIPTION
**Pull request by yuwenzho**
Introduction
[Intel® Extension for Transformers](https://github.com/intel/intel-extension-for-transformers) is an innovative toolkit designed to accelerate GenAI/LLM everywhere with the optimal performance of Transformer-based models on various Intel platforms

Description

adding ITREX runtime embeddings using intel-extension-for-transformers.
added mdx documentation and example notebooks
added embedding import testing.

### CODE REVIEW

**Comment by yuwenzho** [comment]
@baskaryan  Could you please help review this PR? Thank you.

File: docs/docs/integrations/providers/itrex.mdx
Comment (baskaryan) [inline docs/docs/integrations/providers/itrex.mdx:1]: can we add this to the existing intel provider page?

Comment (yuwenzho) [reply docs/docs/integrations/providers/itrex.mdx:1]: Our proposal is to add a new provider page `intel.mdx` to include the contents in both [optimization-intel.mdx](https://github.com/langchain-ai/langchain/blob/master/docs/docs/integrations/providers/optimum_intel.mdx) and itrex.md here. We would love to hear your thoughts.

Comment (baskaryan) [reply docs/docs/integrations/providers/itrex.mdx:1]: a single intel.mdx page in place of the itrex.mdx and optimization-intel.mdx pages would be great

Comment (yuwenzho) [reply docs/docs/integrations/providers/itrex.mdx:1]: I updated the provider mdx. Please review. Thank you.

**Comment by hshen14** [comment]
@baskaryan Please let us know whether the PR is good to be merged. Thanks.

**Review by baskaryan — APPROVED** [review]

**Pull request merged**